File: app/src/main/java/com/omarea/common/ui/BlurTopBarLayout.java
package com.omarea.common.ui;

import android.content.Context;
import android.util.AttributeSet;

public class BlurTopBarLayout extends BlurViewLinearLayout {
    public BlurTopBarLayout(Context context, AttributeSet attrs) {
        super(context, attrs);
        // Giao diện "đảo nổi": giữ nguyên bo góc mặc định (BlurEngine.DEFAULT_CORNER_RADIUS)
        // thay vì ép về hình chữ nhật vuông cạnh như trước, để thanh top bar trông như một
        // khối nổi tách biệt khỏi mép màn hình (kết hợp với margin đặt ở layout XML).
        // Không override drawStroke() nữa: dùng viền bo tròn đầy đủ 4 cạnh của lớp cha
        // (BlurViewLinearLayout) thay vì chỉ vẽ 1 đường kẻ ở cạnh dưới, cho đúng cảm giác
        // "đảo" độc lập có viền bao quanh.
    }
}

File: app/src/main/kotlin/com/omarea/common/ui/BlurTopBarLayout.kt
package com.omarea.common.ui

import android.content.Context
import android.graphics.Canvas
import android.util.AttributeSet

class BlurTopBarLayout(context: Context, attrs: AttributeSet?) : BlurViewLinearLayout(context, attrs) {
    init {
        // Tắt bo góc cho thanh trên (Top Bar thường là hình chữ nhật phẳng)
        this.engine.cornerRadius = 0f
    }

    override fun drawStroke(canvas: Canvas) {
        val paint = BlurEngine.getStrokePaint(context)
        val strokeWidth = paint.strokeWidth

        // GIẢI PHÁP: Tính toán vị trí Y sao cho đường kẻ nằm trọn bên trong View
        // Thay vì vẽ tại getHeight(), chúng ta thụt lên một nửa độ dày của viền
        val y = height - (strokeWidth / 2f)

        // Chỉ vẽ một đường kẻ ngang (Divider) ở cạnh dưới cùng
        canvas.drawLine(0f, y, width.toFloat(), y, paint)
    }
}
